### PR TITLE
Updating the function `compute_apparent_ABmag`

### PR DESCRIPTION
**Pull request by bazkiaei**
This function was one of the inconsistence functions in the code and the goal of this PR is to make it consistent with other functions #70 . Changing this function could cause some effects on other functions and create a need for change to some other functions as well.
The changes that are made here are inspired by the jupyter outflow that is provided in #71 and these changes will be applied to that PR finally.
This is one more step, maybe the most important one, to divide `prepare_mocks` and remove it from the code.

### CODE REVIEW

File: mocks/mocks.py [hunk 1]
```diff
@@ -243,63 +243,63 @@ def scale_light_by_distance(particle_positions,
...
-    total_apparent_ABmag_sim = compute_apparent_ABmag(z,
-                                                      galaxy_mass,
-                                                      mass_to_light,
-                                                      abs_mag_sun,
-                                                      cosmo)
-
-    config['total_mag'] = total_apparent_ABmag_sim
-
-    return config, galaxy_sim_data_raw
+# def prepare_mocks(config):
```
Comment (bazkiaei) [inline mocks/mocks.py:246]: With respect to the jupyter outflow, which is provided in #71 , there will be no need for this function anymore. But I will remove it and its test when this PR is approved.

File: mocks/tests/test_mocks.py [hunk 2]
```diff
@@ -152,43 +152,43 @@ def test_read_gadget(config):
...
+#     assert mocks_config['imager_filter'] == 'g'
+#     assert mocks_config['pixel_scale'].to(u.arcsec / u.pixel).value ==\
+#         pytest.approx(3.506664448419332,
+#                       rel=1e-12)
+#     assert mocks_config['total_mag'].to(u.ABmag).value == pytest.approx(
+#         9.12330433833332,
+#         rel=1e-12)
+
+
+# def test_create_mock_galaxy_noiseless_image(huntsman_sbig_dark_imager,
```
Comment (bazkiaei) [inline mocks/tests/test_mocks.py:168]: With respect to new changes that are made, inputs for this test should be changed. I prefer to make those changes after this PR is approved and in another PR.

**Comment by bazkiaei** [comment]
@AstroHuntsman/spiderlings 
Ready for review.

**Comment by lspitler** [comment]
@bazkiaei suggest renaming to just `compute_apparent_mag` and allow someone to send an argument `mag_units=u.ABmag` that defaults to AB mags.

**Review by fergusL — APPROVED** [review]
Looks good to me

**Comment by bazkiaei** [comment]
> Looks good to me

Thanks, I will wait for approval from @AstroHuntsman/arachne as well.

File: mocks/mocks.py [hunk 3]
```diff
@@ -585,26 +585,27 @@ def compute_apparent_ABmag(z,
...
-    mass_to_light = ensure_unit(mass_to_light, u.M_sun / u.L_sun)
-    abs_mag_sun = ensure_unit(abs_mag_sun, u.ABmag)
+    particle_light = ensure_unit(particle_light, u.L_sun)
     # Total luminosity of the simulated galaxy in solar luminosities,
-    # integrated over the filter band.
-    total_lum_sim = galaxy_mass / mass_to_light
+    # integrated over the demanded filter band.
+    total_lum_sim = particle_light.sum()
     # Total absolute ABmag of the simulated galaxy in the demanded band.
     absolute_ABmag =\
```
Comment (lspitler) [inline mocks/mocks.py:606]:
```suggestion
    absolute_mag =\
```

File: mocks/mocks.py [hunk 4]
```diff
@@ -572,11 +572,11 @@ def compute_total_mass(galaxy_sim_data_raw,
...
-                           abs_mag_sun,
-                           cosmo=WMAP9):
+def compute_apparent_mag(z,
+                         particle_light,
+                         config,
+                         mag_units=u.ABmag,
+                         cosmo=WMAP9):
     """
     This function computes the apparent magnitude of the target galaxy and its
     environment with respect to the demanded distance.
```
Comment (lspitler) [inline mocks/mocks.py:582]:
```suggestion
    environment with respect to a given distance.
```

File: mocks/mocks.py [hunk 5]
```diff
@@ -585,26 +585,27 @@ def compute_apparent_ABmag(z,
     ----------
     z : float
         The redshift of the target galaxy at the demanded distance.
-    galaxy_mass : float
-        The total mass of the target galaxy and all masses in its environment
-        in solar mass units.
-    mass_to_light : float
-        The mass to light ratio of the target.
+    particle_light : astropy.units.Quantity
```
Comment (lspitler) [inline mocks/mocks.py:588]:
```suggestion
    sim_particle_luminosities : astropy.units.Quantity
```

File: mocks/mocks.py [hunk 6]
```diff
@@ -585,26 +585,27 @@ def compute_apparent_ABmag(z,
...
+        The light of the simulation particles.
+    config : dic
+        A dictionary of configuration items.
+    cosmo : astropy.cosmology.core.FlatLambdaCDM, optional
```
Comment (lspitler) [inline mocks/mocks.py:592]: Doesn't this come packed in the `config` dictionary? Suggest removing this.

**Review by lspitler — APPROVED** [review]

File: mocks/mocks.py (hunk 6 above)
Comment (bazkiaei) [reply mocks/mocks.py:592]: No, it does not. But we should make a decision about how to introduce known cosmologies to the code.

**Pull request merged**